packages/ui/src/elements/Thumbnail/index.tsx

```diff
@@ -30,36 +30,56 @@ const ThumbnailContext = React.createContext({
 export const useThumbnailContext = () => React.useContext(ThumbnailContext)
 
 export const Thumbnail: React.FC<ThumbnailProps> = (props) => {
-  const { className = '', doc: { filename } = {}, fileSrc, imageCacheTag, size } = props
+  const { className = '', doc: { filename, mimeType } = {}, fileSrc, imageCacheTag, size } = props
   const [fileExists, setFileExists] = React.useState(undefined)
-
   const classNames = [baseClass, `${baseClass}--size-${size || 'medium'}`, className].join(' ')
+  const [type, setType] = React.useState<'audio' | 'document' | 'image' | 'unknown' | 'video'>(
+    undefined,
+  )
 
   React.useEffect(() => {
-    if (!fileSrc) {
+    if (mimeType) setType(mimeType.split('/')[0])
+  }, [mimeType])
```

Comment:
I dont think this variable really is going to make a huge difference, I would probably move it out of state and out of the effect.

You could just define it inline:

```tsx
const fileType: 'audio' | 'document' | 'image' | 'unknown' | 'video' | false = mimeType && mimeType.split('/')[0]
```

Reply:
Yes, no need to statify the fileType. It's fixed.